### PR TITLE
Fix missing localized names on boundaries

### PR DESCRIPTION
**Pull request by iandees**
- [x] Update tests
- [x] Update data migrations
- [x] Update docs

Connected to #1016.

### CODE REVIEW

File: queries/boundaries.jinja2 [hunk 1]
```diff
@@ -60,6 +60,7 @@ WHERE
 
 SELECT
   tags->'name' AS name,
+  'openstreetmap.org' AS source,
```
Comment (nvkelso) [inline queries/boundaries.jinja2]: Nice catch!

Please also specify Natural Earth as a source at the top of the file in the macro:

```
{% macro ne_boundary_cols() %}
     gid AS __id__,
     'naturalearthdata.com' AS source,
     {% filter geometry %}the_geom{% endfilter %} AS __geometry__,
     mz_calculate_json_boundaries(t.*) AS mz_properties
 {% endmacro %}
```

**Comment by iandees** [comment]
The tests appear to be failing because the right- and left-handedness of the New York/New Jersey boundary is flipped again. Is that arbitrary?

**Comment by zerebubuth** [comment]
The code is [intended to work from a stable orientation](https://github.com/tilezen/vector-datasource/blob/f151b30ef8bd3ffe143a9313e204e2ceb005567a/queries/boundaries.jinja2#L65-L69) but it looks like the post-processor is [sensitive to feature order](https://github.com/tilezen/vector-datasource/blob/f151b30ef8bd3ffe143a9313e204e2ceb005567a/vectordatasource/transform.py#L1774). I suspect that the left/right are getting flipped because the linestring direction is being flipped - so I think it would still display correctly, which might explain why we haven't noticed this problem until now.

Should we enforce a given ordering, perhaps by sorting on `fid` or `tags['name']` before [that loop in the transform](https://github.com/tilezen/vector-datasource/blob/f151b30ef8bd3ffe143a9313e204e2ceb005567a/vectordatasource/transform.py#L1774)?

**Comment by iandees** [comment]
What the heck? Now the tags aren't there at all?

**Comment by zerebubuth** [comment]
It looks like there was an error around the boundaries test, and it logged this:

```
./scripts/../integration-test.py:148: UnicodeWarning: Unicode unequal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  elif v != exp_v:
```

So perhaps they are there, and the unicode unequal comparison means it's picking up the maritime boundary as the "best match", which wouldn't have the same left/right.

**Comment by iandees** [comment]
That should have been fixed in 39797ff (part of this branch) – I wonder what broke. I'll keep looking locally.

**Comment by iandees** [comment]
This still behaves differently on my local instance, but appears to pass here now.

**Comment by iandees** [comment]
(By "differently" I mean that the New York and New Jersey left/right are flipped.)

**Comment by zerebubuth** [comment]
Are you running the CI process locally (i.e: getting data from Overpass), or testing against a metro (or other) extract? Would it be possible to grab the GeoJSON for the tile on your local machine and compare it to dev and see if it's just the tags which are flipped or the direction of the linestring too?

**Comment by iandees** [comment]
After discussing in person, I'm going to add a sort as described in https://github.com/tilezen/vector-datasource/pull/1022#issuecomment-245256220 so that we get consistent results.

**Comment by iandees** [comment]
Whew, finally got it working.

selfie-0

**Comment by nvkelso** [comment]
@zerebubuth Please review :)

**Comment by zerebubuth** [comment]
Looks good to me :+1:

**Pull request merged**